pkg/reconciler/pipelinerun/pipelinespec/pipelinespec.go

```diff
@@ -40,6 +41,19 @@ func GetPipelineData(ctx context.Context, pipelineRun *v1beta1.PipelineRun, getP
 	case pipelineRun.Spec.PipelineRef != nil && pipelineRun.Spec.PipelineRef.Name != "":
 		// Get related pipeline for pipelinerun
 		p, source, err := getPipeline(ctx, pipelineRun.Spec.PipelineRef.Name)
+		if verificationErr, ok := err.(*trustedresources.VerificationError); ok {
+			if trustedresources.IsVerificationError(err) {
+				return nil, nil, verificationErr
+			}
+			pipelineMeta = p.PipelineMetadata()
+			pipelineSpec = p.PipelineSpec()
+			configSource = source
+			return &resolutionutil.ResolvedObjectMeta{
+				ObjectMeta:   &pipelineMeta,
+				ConfigSource: configSource,
+			}, &pipelineSpec, verificationErr
```

Comment:
Why do we need to do this here? If this is an error, do we need to do this? If it's not an error, can we fallthrough to the non-error case below?

Reply:
If it is error, we will return at line 45.
This handles the case where we return a `VerificationError`, but it is warning/pass